scripts/webflow-api-sync.ts

```diff
@@ -0,0 +1,180 @@
+
+const apiItemsBySlug = apiItems.reduce<Record<string, CollectionItem>>((acc, item) => {
+    acc[item.fieldData.slug] = item;
+
+    return acc;
+}, {});
+
+const seen: string[] = [];
+for (const [slug, provider] of Object.entries(neededProviders)) {
+    seen.push(slug);
+
+    if (apiItemsBySlug[slug]) {
+        const item = apiItemsBySlug[slug];
+        if (!item.id) {
+            throw new Error(`Missing item id for ${slug}`);
+        }
+
+        const previous = {
+            fieldData: {
+                name: item.fieldData.name,
+                documentation: item.fieldData['documentation'],
+                'api-categories': item.fieldData['api-categories']
+            }
+        };
+
+        const providerCategories: string[] = provider.categories || [];
+        const apiCategories = providerCategories.map((category) => categoriesBySlug[category]?.id);
+
+        const update = {
+            fieldData: {
+                name: provider.display_name,
+                documentation: provider.url,
+                'api-categories': apiCategories
+            }
+        };
+
+        if (!util.isDeepStrictEqual(previous, update)) {
+            try {
+                await webflow.collections.items.updateItem(apiCollectionId, item.id, update);
+                console.log(`Updated ${slug}`);
+                await setTimeout(rateLimitSleep);
+            } catch (err) {
+                console.error(`Failed to update ${slug}`, err);
+                process.exit(1);
+            }
+        }
+    } else {
+        try {
+            const providerCategories: string[] = provider.categories || [];
+
+            await webflow.collections.items.createItem(apiCollectionId, {
+                fieldData: {
+                    name: provider.display_name,
+                    slug: slug,
+                    documentation: provider.url,
+                    logo: `https://app.nango.dev/images/template-logos/${slug}.svg`,
+                    'api-categories': providerCategories.map((category) => categoriesBySlug[category]?.id)
+                }
+            });
+            console.log(`Created ${slug}`);
+            await setTimeout(rateLimitSleep);
+        } catch (err) {
+            console.error(`Failed to update ${slug}`, err);
+            process.exit(1);
+        }
+    }
+}
+
+const needDeletion = Object.keys(apiItemsBySlug).filter((slug) => !seen.includes(slug));
+for (const toDelete of needDeletion) {
+    try {
+        if (!apiItemsBySlug[toDelete]?.id) {
+            throw new Error('Unexpected missing item id');
+        }
+
+        await webflow.collections.items.deleteItem(apiCollectionId, apiItemsBySlug[toDelete].id);
+        console.log(`Deleted ${toDelete}`);
+        await setTimeout(rateLimitSleep);
+    } catch (err) {
+        console.error(`Failed to delete ${toDelete}`, err);
+        process.exit(1);
+    }
+}
+
+await webflow.sites.publish(siteId, { customDomains: domainIds });
```

Comment:
Every operations (deleteItem, updateItem)  are pending until published?

Reply:
Yes - webflow kind of keeps a next state and then when you publish it posts whatever has changed since the last publish.